### ISSUE TITLE
Possible simplify rule false-negative

### ISSUE THREAD

**Issue by ofek** [astral-sh/ruff#8633]
```python
@cached_property
def python_path(self) -> str:
    if self.name == '3.7':
        if sys.platform == 'win32':
            return r'python\install\python.exe'

        return 'python/install/bin/python3'

    if sys.platform == 'win32':
        return r'python\python.exe'

    return 'python/bin/python3'
```

Is that, with everything selected, supposed to be fine? I would have assumed the return would be flagged as the following recommendation:

```python
return r'python\python.exe' if sys.platform == 'win32' else 'python/bin/python3'
```

Feel free to close if this is the expected behavior!

**Comment by charliermarsh** [comment]
This is expected, but I feel like it could make sense to include under one of the simplify rules? https://docs.astral.sh/ruff/rules/if-else-block-instead-of-if-exp/

**Comment by dhruvmanila** [comment]
(I think there's an open issue with a similar request but I'm unable to find it 😅)
